File: 448-find-all-numbers-disappeared-in-an-array/448-find-all-numbers-disappeared-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    const numCheck = new Set(nums)
    let missing = []
    for(let i = 1; i <= nums.length; i++) {
        if (numCheck.has(i)) {
            continue
        } else {
            missing.push(i)
        }
    }

    return missing;
};

File: 448-find-all-numbers-disappeared-in-an-array/448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int len = std::size(nums);
        std::set<int> numCheck(std::begin(nums), std::end(nums));
        vector<int> missing;
        for (int i = 1; i <= len; ++i)
        {
            if (numCheck.count(i))
            {
                continue;
            }
            else
            {
                missing.insert(missing.end(), i);
            }
        }
        return missing;
    }
};